### PR TITLE
[bug_5790]  Added new functionality: median.

### PR DESCRIPTION
**Pull request by MarcinWyt**
Proszę o review.

https://gist.github.com/MarcinWyt/ab97dda84285c6a46120

https://bugzilla.stxnext.pl/show_bug.cgi?id=5790

### CODE REVIEW

File: src/presence_analyzer/utils.py [hunk 1]
```diff
@@ -162,3 +162,18 @@ def group_start_end_times_by_weekday(items):
         )
 
     return user_week
+
+
+def median(input_list):
+    list_sorted = sorted(input_list)
+    list_size = len(list_sorted)
+    if list_size == 0:
```
Comment (dpolcyn) [inline src/presence_analyzer/utils.py]: ```if not list_size```

File: src/presence_analyzer/templates/base.html [hunk 2]
```diff
@@ -6,7 +6,7 @@
     <meta name="keywords" content="" />
     <meta name="description" content=""/>
     <meta name="author" content="STX Next sp. z o.o."/>
-    <meta name="viewport" content="width=device-width; initial-scale=1.0">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
```
Comment (dpolcyn) [inline src/presence_analyzer/templates/base.html]: You have to change it?

File: src/presence_analyzer/tests.py [hunk 3]
```diff
@@ -346,6 +346,42 @@ def test_mean(self):
...
+        result = utils.median(items)
+        self.assertEqual(result, -23.0)
+
+        items = [-6, -21, -8, -12]
+        result = utils.median(items)
+        self.assertEqual(result, -10.0)
+
+        items = [-1.5, 2.1, -3.1, 4.3]
+        result = int(utils.median(items) * 100000) / 100000.0
+        self.assertEqual(result, 0.3)
```
Comment (dpolcyn) [inline src/presence_analyzer/tests.py]: Read about almostequal.

File: src/presence_analyzer/templates/base.html (hunk 2 above)
Comment (MarcinWyt) [inline src/presence_analyzer/templates/base.html]: No, changed accidently for testing. Orignal line restored.

File: src/presence_analyzer/utils.py [hunk 4]
```diff
@@ -162,3 +162,18 @@ def group_start_end_times_by_weekday(items):
         )
 
     return user_week
+
+
+def median(input_list):
```
Comment (dpolcyn) [inline src/presence_analyzer/utils.py]: Missing doc string.

File: src/presence_analyzer/utils.py [hunk 5]
```diff
@@ -162,3 +162,21 @@ def group_start_end_times_by_weekday(items):
...
+    if not list_size:
+        return 0.0
+    else:
+        if not (list_size % 2):
+            idx1 = list_size / 2 - 1
+            idx2 = idx1 + 1
+            return (list_sorted[idx2] + list_sorted[idx1]) / 2.0
+        else:
+            idx = list_size / 2
+            return list_sorted[idx] * 1.0
```
Comment (mjelonek92) [inline src/presence_analyzer/utils.py]: What's the point of multiplying by 1.0?

Comment (MarcinWyt) [inline src/presence_analyzer/utils.py]: To make sure we'll get a float.

Comment (mjelonek92) [inline src/presence_analyzer/utils.py]: Why is it so important that you get 13.0 instead of 13? In any case you should use ```float(number)```.

**Pull request merged**